src/sshfs_mount/sftp_server.cpp

```diff
@@ -1071,3 +1064,9 @@ int mp::SftpServer::handle_extended(sftp_client_message msg)
 
     return reply_ok(msg);
 }
+
+template <typename T>
+T* multipass::SftpServer::get_handle(sftp_client_message msg)
+{
+    return reinterpret_cast<T*>(sftp_handle(msg->sftp, msg->handle));
```

Comment:
Maybe using `static_cast` which has some compile time checks instead of `reinterpret_cast`.

Reply:
Yup, makes sense.